push_notifications/gcm.py

```diff
@@ -47,11 +52,11 @@ def _gcm_send(data, content_type):
 		"Content-Length": str(len(data)),
 	}
 	request = Request(SETTINGS["GCM_POST_URL"], data, headers)
-	return urlopen(request, timeout=SETTINGS["GCM_ERROR_TIMEOUT"]).read().decode("utf-8")
+	return decodestr(urlopen(request, timeout=SETTINGS["GCM_ERROR_TIMEOUT"]).read())
 
 
-def _fcm_send(data, content_type):
-	key = SETTINGS.get("FCM_API_KEY")
+def _fcm_send(data, content_type, application_id):
+	key = get_gcm_api_key(application_id)
```

Comment:
Yes but if you're sending a notification in bulk to a queryset with a mix of GCM/FCM applications, it's not going to work...
You made a typo it should be `get_fcm_api_key()`.